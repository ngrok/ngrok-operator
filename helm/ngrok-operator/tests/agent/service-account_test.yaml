suite: test agent serviceaccount
templates:
- agent/service-account.yaml
release:
  name: test-release
  namespace: test-namespace
tests:
- it: Should match snapshot
  asserts:
  - matchSnapshot: {}
- it: should create the agent serviceaccount
  asserts:
  - isKind:
      of: ServiceAccount
  - isAPIVersion:
      of: v1
  - hasDocuments:
      count: 1
- it: does not generate a serviceaccount when agent.serviceAccount.create is false
  set:
    agent.serviceAccount.create: false
  asserts:
  - hasDocuments:
      count: 0
- it: Creates a service account using .Values.agent.serviceAccount.name if set
  set:
    agent.serviceAccount.name: test-serviceaccount
  asserts:
  - equal:
      path: metadata.name
      value: test-serviceaccount
