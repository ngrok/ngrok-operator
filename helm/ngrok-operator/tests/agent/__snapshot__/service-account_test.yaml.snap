Should match snapshot:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/component: controller
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: ngrok-operator
        app.kubernetes.io/part-of: ngrok-operator
        app.kubernetes.io/version: 0.14.3
        helm.sh/chart: ngrok-operator-0.17.0-rc.4
      name: test-release-ngrok-operator-agent
      namespace: test-namespace
